TransIT.DAL/Repositories/BaseRepository.cs

```diff
@@ -0,0 +1,49 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.ChangeTracking;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace TransIT.DAL.Repositories
+{
+    public class BaseRepository<TEntity> : IBaseRepository<TEntity> where TEntity : class
+    {
+        protected readonly DbContext _context;
```

Comment:
TransITDBContext

Reply:
No, on my opinion that`s correct, we should take DbSet<> like that - DbContext.Set<> or pass DbSet<> as dependency, we do not need concrete class(TransITDBContext).

Reply:
Actually I am not sure that we need DbContext in Repository, I think the only what we need it is a DbSet<TEntity>